runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -375,19 +375,11 @@ public void prefixTrim(final Token address) {
                 Sleep.sleepUninterruptibly(retryRate);
             } catch (WrongEpochException wee) {
                 long serverEpoch = wee.getCorrectEpoch();
-                // Retry if wrongEpochException corresponds to message epoch (only)
-                if (address.getEpoch() == serverEpoch) {
-                    long runtimeEpoch = runtime.getLayoutView().getLayout().getEpoch();
-                    log.warn("prefixTrim[{}]: wrongEpochException, runtime is in epoch {}, " +
-                            "while server is in epoch {}. Invalidate layout for this client " +
-                            "and retry, attempt: {}/{}", address, runtimeEpoch, serverEpoch, x+1, numRetries);
-                    runtime.invalidateLayout();
-                } else {
-                    // wrongEpochException corresponds to a stale trim address (prefix trim token on the wrong epoch)
-                    log.error("prefixTrim[{}]: stale prefix trim. Prefix trim on wrong epoch {}, " +
-                            "while server on epoch {}", address, address.getEpoch(), serverEpoch);
-                    throw wee;
-                }
+                long runtimeEpoch = runtime.getLayoutView().getLayout().getEpoch();
+                log.warn("prefixTrim[{}]: wrongEpochException, runtime is in epoch {}, while server is in epoch {}. "
```

Comment:
This can be a different epoch than what the layout helped executed with.

Reply:
@Maithem I think they are the same epoch, as this runtime is the same runtime in layoutHelper (they are both using AbstractView's runtime), right?

Reply:
The log statement executes outside the lambda that is executed by the layout helper, so its possible that the layout changes in between.

Reply:
I see, thanks. @Maithem @zalokhan Also I think inside layoutHelper, when we get an exception, we can log the name of the function that is executed together with the details, so that outside the lambda, we don't need to log again the exact details (for example in this case, we just need to log the exception name and  number of retries here, the actual value of correct epoch will be logged in layoutHelper together with the function name: prefixTrim).

Reply:
@Maithem The second parameter in the layoutHelper is true - rethrow all exceptions.
This means there are no retries if WrongEpochException is encountered.

Although there exists a case where if we encounter a ServerNotReadyException first, then there can be a retry.

Reply:
That still doesn't prevent other threads from invalidating the layout between the time the exception happens and this log statement.